File: RepoTransBench/target_projects/Python/C++/djui_alias-tips/tests/original/test_alias_tips.cpp
#include <gtest/gtest.h>
#include <optional>
#include "alias_tips.h"

using namespace alias_tips;

TEST(TestAliasTips, SuggestAliasKnown) {
    EXPECT_EQ(suggest_alias("list"), std::optional<std::string>("ls"));
    EXPECT_EQ(suggest_alias("remove"), std::optional<std::string>("rm"));
    EXPECT_EQ(suggest_alias("copy"), std::optional<std::string>("cp"));
    EXPECT_EQ(suggest_alias("move"), std::optional<std::string>("mv"));
    EXPECT_EQ(suggest_alias("make directory"), std::optional<std::string>("mkdir"));
}

TEST(TestAliasTips, SuggestAliasNone) {
    EXPECT_EQ(suggest_alias("unknown"), std::nullopt);
    EXPECT_EQ(suggest_alias(""), std::nullopt);

    // None (nullptr input): C++ version is not possible as overload, simulate via explicit null.
    // We will test with std::nullptr_t, which hits the template<> version.
    EXPECT_EQ(suggest_alias(nullptr), std::nullopt);

    EXPECT_EQ(suggest_alias(123), std::nullopt);
    std::vector<int> vec;
    EXPECT_EQ(suggest_alias(vec), std::nullopt);
}

TEST(TestAliasTips, IsAliasRecommendedTrue) {
    EXPECT_TRUE(is_alias_recommended("list"));
    EXPECT_TRUE(is_alias_recommended("remove"));
    EXPECT_TRUE(is_alias_recommended("copy"));
    EXPECT_TRUE(is_alias_recommended("move"));
    EXPECT_TRUE(is_alias_recommended("make directory"));
}

TEST(TestAliasTips, IsAliasRecommendedFalse) {
    EXPECT_FALSE(is_alias_recommended("something else"));
    EXPECT_FALSE(is_alias_recommended(""));
    EXPECT_FALSE(is_alias_recommended(nullptr));
    EXPECT_FALSE(is_alias_recommended(123));
    std::vector<int> vec;
    EXPECT_FALSE(is_alias_recommended(vec));
}

File: RepoTransBench/target_projects/Python/Rust/djui_alias-tips/tests/original/test_alias_tips.rs
// Rust translation of test_alias_tips.py (original/unrestricted tests)

use alias_tips::{suggest_alias, is_alias_recommended, CommandArg};

#[cfg(test)]
mod tests {
    use super::*;

    #[test]
    fn test_suggest_alias_known() {
        assert_eq!(suggest_alias("list"), Some("ls"));
        assert_eq!(suggest_alias("remove"), Some("rm"));
        assert_eq!(suggest_alias("copy"), Some("cp"));
        assert_eq!(suggest_alias("move"), Some("mv"));
        assert_eq!(suggest_alias("make directory"), Some("mkdir"));
    }

    #[test]
    fn test_suggest_alias_none() {
        assert_eq!(suggest_alias("unknown"), None);
        assert_eq!(suggest_alias(""), None);

        // Simulate None/null - use Option::<&str>::None
        let none_arg: Option<&str> = None;
        assert_eq!(suggest_alias(none_arg), None);

        // Non-str input: integer
        assert_eq!(suggest_alias(123), None);

        // Non-str input: Vec (to mimic Python list)
        let dummy: Vec<&str> = vec!["a", "b"];
        assert_eq!(suggest_alias(dummy), None);
    }

    #[test]
    fn test_is_alias_recommended_true() {
        assert!(is_alias_recommended("list"));
        assert!(is_alias_recommended("remove"));
        assert!(is_alias_recommended("copy"));
        assert!(is_alias_recommended("move"));
        assert!(is_alias_recommended("make directory"));
    }

    #[test]
    fn test_is_alias_recommended_false() {
        assert!(!is_alias_recommended("something else"));
        assert!(!is_alias_recommended(""));

        let none_arg: Option<&str> = None;
        assert!(!is_alias_recommended(none_arg));

        assert!(!is_alias_recommended(123));

        let dummy: Vec<&str> = vec![];
        assert!(!is_alias_recommended(dummy));
    }
}